File: Week11/P2606.java
import java.util.*;

public class P2606{

    //그래프 저장 2차원 배열 
    static List<List<Integer>> graph;
    static boolean[] visited_dfs;
    static int count=0;

    //바이러스 감염 컴퓨터 개수 찾기 -> dfs로 
    public static void worm_dfs(int v){

        visited_dfs[v] = true;

        for (int i: graph.get(v)) {
            if(visited_dfs[i] == false){
                count++;
                worm_dfs(i);
            }
            
        }

    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        //n은 컴퓨터 개수, m은 네트워크 개수 
        int n = sc.nextInt();
        int m = sc.nextInt();

        //방문 여부 기록 배열 
        visited_dfs = new boolean[n+1];

        //null인 상태에서 에러가 발생하는 것 방지 하기 위해! 
        graph = new ArrayList<>();
        
        for(int i = 0; i <= n; i++){
            graph.add(new ArrayList<>());
        }

        for(int i = 0; i < m; i ++){
            int a = sc.nextInt();
            int b = sc.nextInt();

            //양방향 연결 
            graph.get(a).add(b);
            graph.get(b).add(a);

        }

        worm_dfs(1);
        System.out.println(count);

    }
}

File: Week11/P2606.py

#시작점은 항상 1
start = 1
count = 0

def worm_dfs(graph, visited_dfs, start):

    visited_dfs[start] = True #시작점 방문 기록하기 
    global count
    for v in graph[start]:
        #방문한 적이 없는 경우 
        if (visited_dfs[v] == False):
            count += 1
            worm_dfs(graph, visited_dfs, v)
    

# n은 컴퓨터 수, m은 네트워크 수 
n = int(input())
m = int(input())

#방문 여부 기록
visited_dfs = [False]*(n+1)

#그래프 저장 
graph = [[] for _ in range(n+1)]

#양방향 연결 
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)


worm_dfs(graph, visited_dfs, start)
print(count)